src/handlers/rfqm_handlers.ts

```diff
@@ -0,0 +1,63 @@
+// tslint:disable:max-file-line-count
+import { BigNumber } from '@0x/utils';
+import * as express from 'express';
+import * as HttpStatus from 'http-status-codes';
+
+import { CHAIN_ID } from '../config';
+import { schemas } from '../schemas';
+import { RfqmService } from '../services/rfqm_service';
+import { FetchIndicativeQuoteParams } from '../types';
+import { schemaUtils } from '../utils/schema_utils';
+import { findTokenAddressOrThrowApiError, isETHSymbolOrAddress } from '../utils/token_metadata_utils';
+
+export class RfqmHandlers {
+    constructor(public readonly _rfqmService: RfqmService) {}
+
+    public async getIndicativeQuoteAsync(req: express.Request, res: express.Response): Promise<void> {
+        const params = parseFetchIndicativeQuoteParams(req);
+        const indicativeQuote = await this._rfqmService.fetchIndicativeQuoteAsync(params);
+        res.status(HttpStatus.OK).send(indicativeQuote);
+    }
+}
+
+const parseFetchIndicativeQuoteParams = (req: express.Request): FetchIndicativeQuoteParams => {
+    // HACK - reusing the validation for Swap Quote as the interface here is a subset
+    schemaUtils.validateSchema(req.query, schemas.swapQuoteRequestSchema as any);
+    const apiKey = req.header('0x-api-key');
+    if (apiKey === undefined) {
+        throw new Error('Must access with an API key');
```

Comment:
How do these error bubble up to the user exactly?

Reply:
discussed error handling strategy offline, will implement more specific errors. HTTP errors at the handler level, business logic errors at the app/service level (which are caught and rethrown at the handler level)